internal/controllers/watch/kind.go

```diff
@@ -125,30 +126,37 @@ func (k *KindWatchController) newResourceWatchController(parent *WatchController
 // buildRequests returns a reconcile request for every binding to this resource kind.
 func (k *KindWatchController) buildRequests(synth *apiv1.Synthesizer, comps ...apiv1.Composition) []reconcile.Request {
 	keys := map[string]struct{}{}
+	reqs := []reconcile.Request{}
 	for _, ref := range synth.Spec.Refs {
 		keys[ref.Key] = struct{}{}
+
+		if ref.Resource.Name == "" {
+			continue // ref does not have an "implicit" binding
+		}
+
+		nsn := types.NamespacedName{Namespace: ref.Resource.Namespace, Name: ref.Resource.Name}
+		req := reconcile.Request{NamespacedName: nsn}
+		if !slices.Contains(reqs, req) {
+			reqs = append(reqs, req)
+		}
 	}
 
-	reqs := []reconcile.Request{}
 	for _, comp := range comps {
 		for _, binding := range comp.Spec.Bindings {
 			if _, found := keys[binding.Key]; !found {
 				continue
 			}
 
 			nsn := types.NamespacedName{Namespace: binding.Resource.Namespace, Name: binding.Resource.Name}
-			var exists bool
-			for _, req := range reqs {
-				if req.NamespacedName == nsn {
-					exists = true
-					break
-				}
-			}
-			if !exists {
-				reqs = append(reqs, reconcile.Request{NamespacedName: nsn})
+			req := reconcile.Request{NamespacedName: nsn}
+			idx := slices.Index(reqs, req)
+			if idx > 0 {
+				reqs[idx] = req // override the ref's name/namespace if set
```

Comment:
So the comp binding takes precedence over the synth implicit binding?

Reply:
Left a response about this on Chih-Sheng's comment.